dockerfile-generator.rb

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env ruby
```

Comment:
there is a lot going on here -- what's the reason behind more than just `Dockerfile` additions?